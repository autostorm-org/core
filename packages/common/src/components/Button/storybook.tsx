import React from "react";
import Button from "./Button";

// This default export determines where you story goes in the story list
export default {
  title: "Button",
  component: Button,
};

export const ButtonStory = () => {
  return <Button>Hello</Button>;
};
