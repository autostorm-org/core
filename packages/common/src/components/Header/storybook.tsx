import React from "react";
import Header from "./Header";
import HeaderLogo from "./HeaderLogo";
import { HeaderLocations } from "./constants";
import HeaderItem from "./HeaderItem";
import Spain from "../../assets/flags/Spain.png";
import Brazil from "../../assets/flags/Brazil.png";
import ResponsiveTemplate from "../ResponsiveTemplate";
import ListFixed from "../ListFixed";
import Icon from "../Icon";
import HeaderItemAbsolute from "./HeaderItemAbsolute";
import { ListFullScreen, ListFullScreenRenderer } from "../List";
import { Menu } from "react-feather";
// This default export determines where you story goes in the story list
export default {
  title: "Header",
  component: Header,
};
declare var require: any;

const logoPath = "./../../assets/autofica-defi-day-logo.svg";

const Desktop = (args) => (
  <Header>
    <HeaderLogo
      logoPath={require("./../../assets/autofica-defi-logo-short-6.svg")}
      itemLocation={HeaderLocations.left}
    />
    <HeaderItem>Loan Manager</HeaderItem>
    <HeaderItem>Whitepaper</HeaderItem>
    <HeaderItem
      dropDown
      sections={[
        {
          links: [
            {
              content: (
                <>
                  <span>br-PT</span>{" "}
                  <Icon
                    src={Brazil}
                    width={23}
                    height={13}
                    alt={"portuguese"}
                  />
                </>
              ),
            },
          ],
        },
      ]}
      Submenu={ListFixed}
    >
      es-ES <Icon src={Spain} width={23} height={13} alt={"espanol"} />
    </HeaderItem>
  </Header>
);

const Mobile = (args) => {
  const [isOpen, setOpen] = React.useState(false);
  return (
    <Header
      submenu={
        isOpen ? (
          <ListFullScreen>
            <ListFullScreenRenderer>Vende</ListFullScreenRenderer>{" "}
            <ListFullScreenRenderer>Contacto</ListFullScreenRenderer>
            <ListFullScreenRenderer>Ayuda</ListFullScreenRenderer>
            <ListFullScreenRenderer>Mi Cuenta</ListFullScreenRenderer>
          </ListFullScreen>
        ) : null
      }
    >
      <HeaderLogo
        logoPath={require("./../../assets/autofica-defi-day-logo.svg")}
        itemLocation={HeaderLocations.center}
        overrides={{ paddingTop: 10, paddingBottom: 10, width: 125 }}
      />
      <HeaderItemAbsolute
        itemLocation={HeaderLocations.right}
        onClick={function () {
          setOpen(!isOpen);
        }}
      >
        <Menu color="white" />
      </HeaderItemAbsolute>
    </Header>
  );
};

const ResponsiveStory = () => (
  <ResponsiveTemplate mobile={Mobile} tablet={Desktop} desktop={Desktop} />
);

export const FirstStory = ResponsiveStory.bind({});

FirstStory.args = {
  /* the args you need here will depend on your component */
};
