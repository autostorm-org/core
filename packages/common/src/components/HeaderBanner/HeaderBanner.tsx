import React from "react";
import { HeaderBanner, BoundedRow } from "@autofica/common";
import { createUseStyles } from "@autofica/common";
import testcargoodcrop from "./testcargoodcrop.png";

const useStyles = createUseStyles((theme: any) => ({
  root: {
    backgroundColor: theme.palette.primary.main,
    flexDirection: "row",
    height: 250,
    alignItems: "flex-end",
    display: "flex",
    "@media (max-width:680px)": {
      display: "none",
    },
  },
  flexContainer: {
    display: "flex",
    alignItems: "center",
    width: 800,
    height: 250,
  },
  headerBannerContainer: {
    position: "relative",
    top: 20,
    width: 450,
    maxWidth: "50%",
    zIndex: 51,

    "@media (max-width:768px)": {
      maxWidth: "45%",
    },

    "@media (max-width:900px)": {
      maxWidth: "38%",
    },
  },

  carImage: {
    // position: "absolute",
    zIndex: 50,
    right: 0,
    // transform: "translate(-15%, 0)",
    maxWidth: 400,
    height: "auto",
    "@media (max-width:768px)": {
      // transform: "translate(0, 0)",
      maxWidth: 380,
      position: "relative",
      right: 15,
    },
  },
}));
// type HeaderBarnnerProps = {};

function HeaderBarnner() {
  //   const styles = useStyles(props);
  const styles = useStyles();

  return (
    <div className={styles.root}>
      <BoundedRow>
        <div className={styles.flexContainer}>
          <div className={styles.headerBannerContainer}>
            <HeaderBanner
              bannerText={"Buy your next car with cryptocurrency."}
            />
          </div>

          <img alt={"Car"} src={testcargoodcrop} className={styles.carImage} />
        </div>
      </BoundedRow>
    </div>
  );
}
export default HeaderBarnner;
