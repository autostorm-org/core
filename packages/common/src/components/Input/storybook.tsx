import React from "react";
import Input from "./Input";
// This default export determines where you story goes in the story list
export default {
  title: "Input",
  component: Input,
};

const Template = () => <Input />;

export const FirstStory = Template.bind({});
