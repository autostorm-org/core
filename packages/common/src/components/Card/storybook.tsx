import React from "react";
import Card from "./Card";
import Typography, { EnumTypographyVariant } from "../Typography";
import Button from "../Button";
import { withKnobs, boolean } from "@storybook/addon-knobs";
import { ThemeContext, nightTheme, defaultTheme } from "../../theme";
import ResponsiveTemplate from "../ResponsiveTemplate";
// This default export determines where you story goes in the story list
export default {
  title: "Card",
  component: Card,
  decorators: [withKnobs],
};

const MobileCard = () => {
  return (
    <div style={{ height: 184, width: 350 }}>
      <Card>
        <Typography variant={EnumTypographyVariant.h3}>Card title</Typography>
        <br />
        <Typography variant={EnumTypographyVariant.p}>
          Card body contents. Card body contents. Card body contents. Card body
          contents.
        </Typography>
        <br />
        <Button onClick={() => {}} overrides={{ marginTop: 9 }}>
          Call to action
        </Button>
      </Card>
    </div>
  );
};

const DesktopCard = () => {
  return (
    <div style={{ height: 240, width: 560 }}>
      <Card style={{ padding: "30px" }}>
        <Typography variant={EnumTypographyVariant.h1}>
          Crypto car loan finder
        </Typography>
        <br />
        <Typography variant={EnumTypographyVariant.p}>
          Get a collateralized loan for your next car purchase. Chose your
          backing assets. Powered by Multiplier.{" "}
        </Typography>
        <br />
        <Button onClick={() => {}} overrides={{ marginTop: 10 }}>
          Call to action
        </Button>
      </Card>
    </div>
  );
};

const ResponsiveStory = () => {
  const isDayMode = boolean("DAY MODE", true);
  return (
    <ThemeContext.Provider value={isDayMode ? defaultTheme : nightTheme}>
      <ResponsiveTemplate
        mobile={MobileCard}
        desktop={DesktopCard}
        tablet={DesktopCard}
      />
    </ThemeContext.Provider>
  );
};

export const FirstStory = ResponsiveStory.bind({});
