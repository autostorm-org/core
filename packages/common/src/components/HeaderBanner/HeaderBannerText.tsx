import React from "react";
import { createUseStyles } from "../../theme";
import { Button } from "../Button";

const useStyles = createUseStyles(function (theme: any) {
  return {
    root: {
      textAlign: "left",
      font: `${theme.headerBanner.fontWeight} ${theme.headerBanner.fontSize}/30px ${theme.headerBanner.font}`,
      letterSpacing: "0px",
      color: theme.headerBanner.color,
      opacity: "1",
      backgroundColor: theme.headerBanner.backgroundColor,
      padding: "18px 36px 18px 26px",
      display: "inline-block",
    },
  };
});

type HeaderBannerTextProps = {
  overrides?: any;
  bannerText: string;
};

const defaultProps: HeaderBannerTextProps = {
  overrides: null,
  bannerText: "hello",
};

function HeaderBannerText(props: HeaderBannerTextProps) {
  const styles = useStyles(props);
  return (
    <div className={styles.root} style={props.overrides}>
      <span>{props.bannerText}</span>
      <br />
      <span>Hello</span>
      <br />

      <Button>Hello</Button>
    </div>
  );
}

HeaderBannerText.defaultProps = defaultProps;
export default HeaderBannerText;
export { HeaderBannerText, HeaderBannerTextProps };
