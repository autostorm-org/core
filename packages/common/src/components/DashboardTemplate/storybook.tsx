import React from "react";
import DashboardTemplate from "./DashboardTemplate";
import { BoundedRow } from "../BoundedRow";
import { Typography, TypographyVariant_ENUM } from "../Typography";
import Icon from "../Icon";
import Spain from "../../assets/flags/Spain.png";
import { HeaderResponsive } from "../Header";
import DashboardOptionsColumn from "./DashboardOptionsColumn";
import DashboardOptionsItem from "./DashboardOptionsItem";
import { PillButton } from "./../PillButton";
import { Eye, ArrowLeftCircle, DollarSign, ArrowUp, Key } from "react-feather";

export default {
  title: "Dashboard Template Stories",
};

export const DashboardOptionsColumnStory = () => {
  return (
    <DashboardOptionsColumn>
      <DashboardOptionsItem activeElement href={"#"}>
        hello now
      </DashboardOptionsItem>
      <DashboardOptionsItem href={"#"}>hello now</DashboardOptionsItem>
      <DashboardOptionsItem href={"#"}>hello now</DashboardOptionsItem>
    </DashboardOptionsColumn>
  );
};

export const DashboardTemplateStory = () => {
  return (
    <>
      <HeaderResponsive
        options={[
          { content: "Loan Manager", href: "#" },
          { content: "Whitepaper", href: "#" },
          {
            content: (
              <>
                es-ES{" "}
                <Icon src={Spain} width={23} height={13} alt={"espanol"} />
              </>
            ),
          },
        ]}
      />
      <div style={{ marginTop: 30 }}>
        <BoundedRow>
          <DashboardTemplate
            mainContent={
              <Typography
                variant={TypographyVariant_ENUM.h3}
                style={{ padding: 30 }}
              >
                Main Content
              </Typography>
            }
            optionsContent={
              <>
                <DashboardOptionsColumn style={{ borderTopLeftRadius: 8 }}>
                  <div style={{ marginTop: 30, marginBottom: 20 }}>
                    <PillButton
                      onClick={() => {
                        console.log("CLIICK");
                      }}
                    >
                      <Icon
                        src={"crypto-logos/bsc.svg"}
                        height={22}
                        width={22}
                        alt={"BSC logo"}
                        style={{ marginRight: 10 }}
                      />
                      BSC Mainnet
                    </PillButton>
                  </div>

                  <DashboardOptionsItem href={"#"} activeElement>
                    <Eye style={{ marginRight: 10 }} />
                    Loans Overview
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <ArrowLeftCircle style={{ marginRight: 10 }} />
                    Borrow
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <DollarSign style={{ marginRight: 10 }} />
                    Payback
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <ArrowUp style={{ marginRight: 10 }} />
                    Withdraw Collateral
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <Key style={{ marginRight: 10 }} />
                    Deposit
                  </DashboardOptionsItem>
                </DashboardOptionsColumn>
              </>
            }
          />
        </BoundedRow>
      </div>
    </>
  );
};
