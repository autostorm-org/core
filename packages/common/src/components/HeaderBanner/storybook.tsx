import React from "react";
import HeaderBannerText from "./HeaderBannerText";
// This default export determines where you story goes in the story list
export default {
  title: "HeaderBannerText",
  component: HeaderBannerText,
};

const Template = (args) => <HeaderBannerText>Hello</HeaderBannerText>;

export const FirstStory = Template.bind({});

FirstStory.args = {
  /* the args you need here will depend on your component */
};
