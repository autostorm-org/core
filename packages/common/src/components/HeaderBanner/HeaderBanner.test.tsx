import React from "react";
import renderer from "react-test-renderer";
import { FirstStory } from "./HeaderBanner.storybook";
describe("HeaderBanner Test", () => {
  it("Should render HeaderBanner component", () => {
    const root = renderer.create(<FirstStory />);
    let tree = root.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
