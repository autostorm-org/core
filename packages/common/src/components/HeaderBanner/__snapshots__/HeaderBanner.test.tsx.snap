// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBanner Test Should render HeaderBanner component 1`] = `
<div
  className="root"
>
  <div
    className="root  "
  >
    <div
      className="flexContainer"
    >
      <div
        className="headerBannerCardWrapper"
      >
        <div
          className="headerBannerContainer"
        >
          <div
            className="root bannerCardOverride"
          >
            <span
              className="h1 titleOverride"
            >
              Crypto car loan finder
            </span>
            <br />
            <span
              className="p"
            >
              Get a collateralized loan for your next car purchase. Chose your backing assets. Powered by Multiplier.
            </span>
            <br />
            <button
              className="root buttonOverride"
              disabled={false}
            >
              Find your car loan
            </button>
          </div>
        </div>
      </div>
      <div
        className="carImage"
      >
        <img
          alt="Car illustration"
          className="headerBannerImage"
          loading="lazy"
          src="./carillustration.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
